core/request.go

```diff
@@ -175,5 +178,31 @@ func (r *RequestAccessor) ProxyEventToHTTPRequest(req events.APIGatewayProxyRequ
 	httpRequest.Header.Add(APIGwContextHeader, string(apiGwContext))
 	httpRequest.Header.Add(APIGwStageVarsHeader, string(stageVars))
 
+	lc, _ := lambdacontext.FromContext(ctx)
+	rc := requestContext{lambdaRuntime: lc, gatewayProxy: req.RequestContext}
+	ctx = context.WithValue(httpRequest.Context(), requestContextKey,
+		rc)
+	httpRequest = httpRequest.WithContext(ctx)
 	return httpRequest, nil
 }
+
+// GetAPIGatewayContextFromContext retrieve APIGatewayProxyRequestContext from context.Context
+func GetAPIGatewayContextFromContext(ctx context.Context) (events.APIGatewayProxyRequestContext, bool) {
+	v, ok := ctx.Value(requestContextKey).(requestContext)
+	return v.gatewayProxy, ok
+}
+
+// GetRuntimeContextFromContext retrieve Lambda Runtime Context from context.Context
+func GetRuntimeContextFromContext(ctx context.Context) (*lambdacontext.LambdaContext, bool) {
+	v, ok := ctx.Value(requestContextKey).(requestContext)
+	return v.lambdaRuntime, ok
+}
+
+type key struct{}
```

Comment:
for clarity, I think this should called `ctxKey`

Reply:
Renamed